### ISSUE TITLE
Filtering on Relay with ForeignKey Models

### ISSUE THREAD

**Issue by hasyemiraws** [graphql-python/graphene-django#767]
When I try using the filter class for related models using ForeignKey, I found a problem when using the filter_fields with Relay.

So I have two models, **Category** and **CategoryTranslation**: 

```
class Category(models.Model):
   active = models.BooleanField(default=True)
   products = models.ManyToManyField(
        'Product', related_name='category_products', through='ProductCategory')
   date_add = models.DateTimeField(auto_now_add=True)
   date_upd = models.DateTimeField(auto_now=True)
```

```
class CategoryTranslation(SeoModelTranslation):
    slug = models.SlugField(max_length=256, allow_unicode=True)
    name = models.CharField(max_length=300, default=False)
    lang = models.CharField(max_length=10)
    category = models.ForeignKey(
        Category, related_name='translations', on_delete=models.CASCADE)
```  

On my schema, I add translations to the filter_field like this:

```
class Category(DjangoObjectType):
    class Meta:
        interfaces = (graphene.relay.Node, )
        model = models.Category
        filter_fields = {
            'id': ['exact', 'istartswith'],
            'translations__name': ['exact', 'icontains', 'in'],
            'translations__lang': ['exact']
        }
        connection_class = CountableConnection
```

But the result that I got when I use icontains are wrong, cause it's showing me duplicate result. (Basically it will show the category times the number of the translations.

Sample result:
```
{
  "data": {
    "categories": {
      "edges": [
        {
          "node": {
            "id": "Q2F0ZWdvcnk6Nw==",
            "active": true,
            "translations": {
              "edges": [
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjoxNg==",
                    "lang": "es",
                    "name": "es test"
                  }
                },
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjoxNA==",
                    "lang": "en",
                    "name": "test en"
                  }
                },
                }
              ]
            }
          }
        },{
          "node": {
            "id": "Q2F0ZWdvcnk6Nw==",
            "active": true,
            "translations": {
              "edges": [
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjoxNg==",
                    "lang": "es",
                    "name": "es test"
                  }
                },
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjoxNA==",
                    "lang": "en",
                    "name": "test en"
                  }
                },
                }
              ]
            }
          }
        }, {
          "node": {
            "id": "Q2F0ZWdvcnk6MTI=",
            "active": true,
            "translations": {
              "edges": [
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjo2Mg==",
                    "lang": "es",
                    "name": "another category test es"
                  }
                },
                {
                  "node": {
                    "id": "Q2F0ZWdvcnlUcmFuc2xhdGlvbjo2Mg==",
                    "lang": "en",
                    "name": "another category test en"
                  }
                },
               }
              ]
            }
          }
        }
      ]
    }
  }
}
```

So for example in the result, the first and the second is the same..

My sample query for that is like this: 

```
query allCategories {
  categories(first: 10, translations_Name_Icontains: "test") {
    edges {
      node {
        ........
      }
    }
  }
}
```

What is it that I do wrong here?

**Comment by zbyte64** [comment]
#758 talks about a bug where self-joining occurs when using `DjangoFilterConnectionField` , I wonder if that is ocuring here?

**Comment by jkimbo** [comment]
This should be fixed in [v2.7.0](https://github.com/graphql-python/graphene-django/releases/tag/v2.7.0)

**Comment by silvanm** [comment]
@hasyemiraws, I am experiencing the same in #1051. Has 2.7.0  fixed it for you, as suggested by @jkimbo?
